### PR TITLE
Fix Watcher email actions not using WebhookService for HTTP requests

### PR DESCRIPTION
**Pull request by dakrone**
This commit fixes a bug where the internal `WebhookService` was not being used for Watcher's `email` attachment `reporting` or `url` types. This meant that why the additional token configured as part of #93426 was sent for `webhook` Watcher actions, it was not being sent in the HTTP request made when resolving the attachments of the `email` action.

To accomplish this, the parts of `WebhookService` that modify and make the request have been extracted, and the `WebhookService` is now used by the `ReportingAttachmentParser` and `HttpEmailAttachmentParser` to perform their HTTP requests.

Additionally, some debug logging has been added for when the token is sent (and when a `WebhookService` request is executed).

### CODE REVIEW

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/WebhookService.java [hunk 1]
```diff
@@ -121,41 +122,80 @@ public Action.Result execute(
...
             return new WebhookAction.Result.Success(redactToken.apply(request), response);
         }
     }
 
+    /**
+     * Makes any additional modifications to the {@link HttpRequest} if necessary.
+     * If no modifications are made the same instance is returned, otherwise a new
+     * HttpRequest is returned.
+     */
+    private HttpRequest maybeModifyHttpResponse(HttpRequest request) {
```
Comment (masseyke) [inline x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/WebhookService.java:159]: This ought to be `maybeModifyHttpRequest` right? It's not modifying the response.

**Review by masseyke — COMMENTED** [review]
The code makes sense and looks good to me (aside from the one method name), but we probably ought to have a test that asserts that it's now actually modifying the request. I don't think any of them currently cover that, do they?

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/WebhookService.java (hunk 1 above)
Comment (dakrone) [reply x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/WebhookService.java:159]: Yep, that's a typo, I pushed a commit to fix this.

**Review by masseyke — APPROVED** [review]
LGTM

**Pull request merged**